modules/sandbox.js

```diff
@@ -121,19 +123,70 @@ function createSandbox(aScript, aContentWin, aUrl, aFrameScope) {
         'contentStartRequest');
   }
 
-  // TODO: lazy getter?
-  Components.utils.evalInSandbox(
-      'const GM_info = ' + uneval(aScript.info()), sandbox);
+  injectGMInfo(aScript, sandbox);
 
   return sandbox;
 }
 
 
+
+function injectGMInfo(aScript, sandbox) {
+  var rawInfo = aScript.info();
+  var scriptURL = aScript.fileURL;
+
+  // TODO: also delay top level clone via lazy getter? XPCOMUtils.defineLazyGetter
+  sandbox.GM_info = Cu.cloneInto(rawInfo, sandbox);
+  
+  var waivedInfo = Components.utils.waiveXrays(sandbox.GM_info);
+  
+  var fileCache = new Map();
+
+  // lazy getters for heavyweight strings that aren't sent down through IPC
+  Object.defineProperty(waivedInfo,
+    "scriptSource",
+    { get: Cu.exportFunction(
+      function() {
+        var content = fileCache.get("scriptSource");
+        if(content === undefined) {
+          content = GM_util.fileXHR(scriptURL, "application/javascript");
+          fileCache.set("scriptSource", content);
+        }
+        return content;
+      }, sandbox)
+    }
+  );
+  
+  // meta depends on content, so we need a lazy one here too
+  Object.defineProperty(waivedInfo,
+    'scriptMetaStr',
+    {get: Cu.exportFunction(
+      function() {
+        var meta = fileCache.get("meta");
+        if(meta === undefined) {
+          meta = extractMeta(this.scriptSource);
```

Comment:
It throws the error:
![1](https://cloud.githubusercontent.com/assets/2373486/9838787/bb53ae82-5a65-11e5-8270-3f2090c3b2d6.png)